dapp/src/components/buySell/SellWidget.js

```diff
@@ -221,10 +219,12 @@ const SellWidget = ({
         result = await vaultContract.redeemAll(minStableCoinsReceivedBN, {
           gasLimit,
         })
-        storeTransaction(result, `redeem`, returnedCoins, coinData)
+        receipt = await rpcProvider.waitForTransaction(result.hash)
+        const data = await getStableCoinLogs(receipt)
+
+        storeTransaction(result, `redeem`, returnedCoins, data)
```

Comment:
If I understand correctly this change would prevent transactions from showing up in the sidebar while they are not yet mined. We show transaction as soon as it is submitted, and update it once it is mined.

See [TransactionListener](https://github.com/OriginProtocol/origin-dollar/blob/6e2b1738fe876aa8e7e6d38c23417383ed33b3ad/dapp/src/components/TransactionListener.js) that is taking care of updating mined transactions.

TransactionListener would probably also be a better place where to populate the `getStableCoinLogs`

Reply:
Yes you're correct. in transactionListener, we can take care of every transactions whatever its type is. And that way, we can make it simple and also high performance as well. since in the transactionListener, it originally calling `waitForTransaction` method so we don't need to call this function twice.
But what I'm concerned is the transaction logs which are returned from `waitForTransaction` seems not correct on my local network. there is no stable coin addresses listed in our system at least.
Is it expected issue?